### PR TITLE
FocusZone: Using keyboard-key library in v7 to align it closer to v0

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

This PR replaces the references to `ev.which` and `KeyCodes` in v7's version of `FocusZone` in favor of the `keyboard-key` library that is being used already in v0. This allows us to bring the two versions closer together.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12270)

### CODE REVIEW

File: packages/react-focus/package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "@uifabric/set-version": "^7.0.6",
     "@uifabric/styling": "^7.10.17",
     "@uifabric/utilities": "^7.13.2",
+    "keyboard-key": "1.0.1",
```
Comment (ecraig12345) [inline packages/react-focus/package.json]: You should update everyone to 1.1.0 which has types. Trouble is that the types only work properly if you have `esModuleInterop` enabled, which fluent doesn't (Levi or Shift could give details of why). The current typing format is necessary to properly reflect the code's assignment to `module.exports`...though come to think of it, you could just update keyboard-key to a more interop-friendly export format. It's here: https://github.com/levithomason/keyboard-key/blob/master/src/keyboardKey.js

**Comment by dzearing** [comment]
OOF on bundle size...

File: packages/react-focus/package.json (hunk 1 above)
Comment (khmakoto) [reply packages/react-focus/package.json]: I agree, we should update everyone. @layershifter and @levithomason why doesn't Fluent have `esModuleInterop`?

Comment (ecraig12345) [reply packages/react-focus/package.json]: There's more background here (`allowSyntheticDefaultImports` and `esModuleInterop` are closely related): https://github.com/microsoft/fluent-ui-react/pull/1395

**Comment by layershifter** [comment]
I would say that `keyboard-key` should be refactored to ES modules as now it will always import everything...

**Comment by khmakoto** [comment]
Better but still not great:

__Before:__
Before

__After:__
After

**Comment by khmakoto** [comment]
Closing in favor of #12339. I'll copy [Levi's comment](https://github.com/OfficeDev/office-ui-fabric-react/pull/12270#pullrequestreview-377336274) as TODOs in the description there.

**Pull request closed without merging**